feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/my_rated/MyRatedViewModel.kt

```diff
@@ -0,0 +1,130 @@
+package com.chocolatecake.viewmodel.my_rated
+
+import androidx.lifecycle.viewModelScope
+import androidx.paging.CombinedLoadStates
+import androidx.paging.LoadState
+import androidx.paging.cachedIn
+import androidx.paging.map
+import com.chocolatecake.bases.BaseViewModel
+import com.chocolatecake.usecase.my_rated.GetMyRatedMoviesUseCase
+import com.chocolatecake.usecase.my_rated.GetMyRatedTVShowsUseCase
+import com.chocolatecake.viewmodel.common.listener.MovieListener
+import com.chocolatecake.viewmodel.my_rated.mappers.MyRatedMovieToMovieHorizontalUiMapper
+import com.chocolatecake.viewmodel.my_rated.mappers.MyRatedTvShowToMovieHorizontalUiMapper
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class MyRatedViewModel @Inject constructor(
+    private val getRatedTVShowsUseCase: GetMyRatedTVShowsUseCase,
+    private val getRatedMoviesUseCase: GetMyRatedMoviesUseCase,
+    private val myRatedMovieToMovieHorizontalUiMapper: MyRatedMovieToMovieHorizontalUiMapper,
+    private val myRatedTvShowToMovieHorizontalUiMapper: MyRatedTvShowToMovieHorizontalUiMapper,
+) : BaseViewModel<MyRatedUiState, MyRatedEvents>(MyRatedUiState()) , MyRatedListner , MovieListener {
+
+
+    init {
+        getData()
+    }
+
+    private fun getData() {
+        when (_state.value.MyRateType) {
+            RateType.Movies -> getMyRatedMovies()
+            RateType.TVShows -> getMyRatedTvShow()
+        }
+    }
+
+    fun getMyRatedMovies(){
```

Comment:
make this function private

Reply:
i call it in fragment when i click on the chip